qemu: Tweak configuration script for qemu 5.1

amshinde opened the pull request:
Certain config options have been deprecated with 5.1.
Remove these.

Depends-on: github.com/kata-containers/runtime#2993

Fixes #1150

Signed-off-by: Archana Shinde <archana.m.shinde@intel.com>

amshinde commented:
/test-ubuntu

devimc commented on scripts/configure-hypervisor.sh:
```diff
@@ -373,6 +373,10 @@ generate_qemu_options() {
+	fi
+
+
+	if [[ "${qemu_version_major}" -ge 5  && "${qemu_version_minor}" -ge 1 ) ]]; then
```
no - this condition will be false for QEMU 6.0

jodh-intel replied:
Eeek! good catch @devimc!!

amshinde commented:
/test-ubuntu

devimc commented on scripts/configure-hypervisor.sh:
```diff
@@ -373,6 +373,10 @@ generate_qemu_options() {
+	fi
+
+
+	if [ "$(echo "${qemu_version_major}.${qemu_version_minor} >= 5.1" | bc)" == "0" ]; then
```
please add a comment

amshinde commented:
@devimc @jcvenegas Seeing this error in the logs:
```
Successfully built e4a4643eca23
16:39:21 Successfully tagged qemu-static:latest
16:39:25 tar: */bin/qemu-pr-helper: Not found in archive
16:39:25 tar: */bin/virtfs-proxy-helper: Not found in archive
16:39:25 tar: Exiting with failure status due to previous errors
16:39:25 Failed at 41: prefix="${PREFIX}" "${GOPATH}/src/${packaging_repo}/static-build/qemu/build-static-qemu.sh"
16:39:26 Build step 'Execute shell' marked build as failure
16:39:32 Performing Post build task...
```
Looks like qemu built, but seeing this error while generating the tar.
Have you seen that before or have any hints of what is going on?

amshinde committed:
static-build: update qemu blacklist file for qemu 5.1

Remove `qemu-pr-helper` and `virtfs-proxy-helper` from the
blacklist file since these files have been moved from `bin`
to `libexec` directory and we ignore the entire `libexec` dir.

Signed-off-by: Archana Shinde <archana.m.shinde@intel.com>

amshinde commented:
/test-ubuntu

amshinde commented:
/AzurePipelines run

amshinde commented:
/test

amshinde committed:
qemu: add kata patches for QEMU 5.1

Add the following patches for QEMU 5.1:
* memory-backend-file/nvdimm: support read-only files as
* 9p: removing coroutines of 9p to increase the I/O performance

Signed-off-by: Archana Shinde <archana.m.shinde@intel.com>

amshinde commented:
/test

amshinde committed:
qemu: apply upstream qemu v5.1 stable fixes

Qemu v5.1 was released with an affending commit 9b3a35ec82
(virtio: verify that legacy support is not accidentally on).
As a result, it breaks commandline compatiblilities for old qemu
users. Upstream qemu has fixed it but no release has been put out yet.
Let's apply these fixes by hand for now.

Refs: https://www.mail-archive.com/qemu-devel@nongnu.org/msg729556.html

Signed-off-by: Archana Shinde <archana.m.shinde@intel.com>

amshinde commented:
/test

fidencio commented:
Folks, this PR has been stale and I'm closing it as we're walking towards the **final** 1.13.0 release of the kata-containers-1.x.
From now on, all the development will happen as part of https://github.com/kata-containers/kata-containers/ and fixes should be provided to that repo.

Thanks a lot for your contribution, we're sorry that for some reason it got stale, and we hope to see you around for new adventures in the kata-containers-2.x land.
